### ISSUE TITLE
Show size of data group

### ISSUE THREAD

**Issue by janagombitova** [akvo/akvo-lumen#2805]
# Context 
With simplifying the dataset UI we removed a top row that showed the count of total rows and columns per dataset. Since this change, we have received feedback that having a quick overview of the size of the dataset is missing in Lumen and if we can bring the counter back. 

# Why: 

For example, Molo mentions in a support issue (https://akvoo.reamaze.com/admin/conversations/akvolumen-support-request-number-375805-from-andrew-at-akvo-dot-org) that: 
> When you open the dataset tab in Lumen, You can't tell the number of rows and the number of columns present in the dataset. Before any transformation, analysis or visualisation one needs to understand the data he/she is working with. One of the aspects of a dataset is its size. 

Carmen in our research call on the new dataset UI and RQGs also mentioned: 
> Before starting exploring data it is good to know the size of the dataset. The size can quickly tell you what visualisation is best to use, for example. Even more important is to know that some groups have a different size (row count) than others as you will analyse them differently. 

Ilyasse mentioned another reason why the count was useful: 
> After updating a dataset we checked the row count to see that the update succeeded and new data was imported well. Even though you get the confirmation, seeing the number grow validated that indeed it went well.   

# Opportunity 
The idea is to start simple. Can we for example add the count of rows and columns to each data group? 

Maybe in the future, before looking at the grid the user could see an overview dashboard with visuals and summaries of the dataset, but before we take that leap we can add something simple to ease the pain. 

@Kiarii what are your thoughts?

**Comment by Kiarii** [comment]
I think the gist that the col & row count was an idea of how many submissions had been made to a form (the col count only informed how many questions were in the form) - so I think a natural starting point may be to provide the number of submissions to the form; this should (I assume) inform the Salim / Petra / et al whether the amount of submissions made suffice for the analysis they are looking to do.. 
Besides that we also had the idea of providing a col/question summary; starting with a simple textual summary to enable the analyst to understand the data in a col and therefore to be able to determine a course of action..

**Comment by janagombitova** [comment]
@Kiarii can you detail how this could work/show/behave?

> ... I think a natural starting point may be to provide the number of submissions to the form..

**Comment by Kiarii** [comment]
In considering how we might show form submissions, i started tinkering with the dataset header as illustrated below; [prototype](https://www.figma.com/proto/XQtjy781ugZ8dmmaA9hWnO/Lumen-style-guide?node-id=68%3A185&scaling=min-zoom) 
<img width="712" alt="Screenshot 2020-09-16 at 10 54 44" src="https://user-images.githubusercontent.com/12692444/93308872-8355fa00-f80b-11ea-903b-b31a4bc05f7f.png">


This direction of thought would imply some changes as unto how being offline - which had been handled in #2706. The offline mode in dark-demo and akvotest are broken however:

- I couldn't get the offline notification functionality to work 🤔 The 'all changes saved' appears when one is effectively offline.
- Col edit and transform actions are also available and usable, at least once before editing is disabled. This creates room for confusion. It would be easier to simply disable any changes when connection is lost and therefore to remove the thinking that  dataset editing is possible when in reality such edits will not be saved; as the user will discover once they reconnect, and have to be redo them..  

 
If editing is disabled while offline, then we do not to provide a long offline notification "Saving failed. Last attempt x seconds ago. Retry." Essentially we just need to say there is no connection or indicate that Lumen is trying to reconnect - and since this happens automatically, we need not provide a 'Retry' action
<img width="735" alt="Screenshot 2020-09-16 at 11 13 26" src="https://user-images.githubusercontent.com/12692444/93310596-ab465d00-f80d-11ea-92db-0c864946e0cf.png">

**Comment by janagombitova** [comment]
@Kiarii Thank you for the changes! Great ideas. Just to be clear, with this issue we will **only**: 
* move the `all changes saved` next to the title. 
* change from text `all changes saved` to icon. How will we show `saving`? 
* set a limit to how long a title can be (max characters) or handle long titles in a different way --> any ideas? 
* add the count of `form submission` under the title. I assume this will represent the number of rows in the Metadata group for Flow datasets (as other groups might have more rows). 

Changing the offline/online indication will be handled in another issue.

**Comment by Kiarii** [comment]
- The saving, I note would be a new state, atm we dont show saving. We could use a cloud-syncing icon to indicate that;
<img width="469" alt="Screenshot 2020-09-18 at 14 08 53" src="https://user-images.githubusercontent.com/12692444/93591417-fd77b180-f9b8-11ea-9075-a5e1bbdbd8da.png">
 
- About the form submission count, Yes
- For the asset title & subtitle we should aim for a smaller font size but with more weight - that shd give us room to accommodate longer titles..
- I already created #2898 for the connection state indication

**Comment by janagombitova** [comment]
Perfect. Let's give it a go. Please note when implementing that we showed the count of rows in the UI before: https://user-images.githubusercontent.com/12456965/83027432-6913c880-a030-11ea-8c8a-b44797d3bae1.png

**Comment by janagombitova** [comment]
Works well. Great job! 

Here are a few examples 

<img width="1276" alt="Screen Shot 2020-10-15 at 19 52 06" src="https://user-images.githubusercontent.com/12456965/96167753-23dd2e00-0f20-11eb-811c-9adbc8f5c1ae.png">
<img width="1280" alt="Screen Shot 2020-10-15 at 19 51 53" src="https://user-images.githubusercontent.com/12456965/96167692-1162f480-0f20-11eb-9910-37b13431967d.png">
<img width="1280" alt="Screen Shot 2020-10-15 at 19 52 28" src="https://user-images.githubusercontent.com/12456965/96167707-1627a880-0f20-11eb-996a-86638487372f.png">

**Comment by janagombitova** [comment]
With @kardan we spoke this morning on how to improve this implementation as we use the word `submissions` also for datasets coming from csv files and links and the count is not correct. We agreed that: 
* for csv file and link datasets we will use the word `rows` as `submissions` makes sense in the context of Flow datasets, especially if the have RQG data where the row count differs per group, but it does not make sense for external datasets.
* make sure the count shows the correct value

**Comment by janagombitova** [comment]
Closing this issue as it will be handled in #2996
